### ISSUE TITLE
goTenna emergency communications improvement

### ISSUE THREAD

**Issue by TechExaminer** [gotenna/PublicSDK#33]
Someone needs to utilize the SDK’s from Gotenna, FireChat, & Bridgefy to create a gateway app that links communications between and across these 3 mesh networks. That way people who can’t afford to buy a Gotenna can download a free mesh app like Firechat or Bridgefy and still have the potential to take advantage of the goTenna Mesh network during an emergency.

**Comment by NF8C** [comment]
I was with you on the Bridgefy until I checked it out.  Seems to require licensing on a per end user basis.  No prices are listed publicly.

**Comment by TechExaminer** [comment]
How about Firechat instead then? Or if that’s not doable then maybe a fresh new app that utilizes Apple’s Multipeer Connectivity framework to create a Bluetooth/WiFi Mesh network chat app similar to Bridgefy or Firechat but also utilizing the goTenna SDK to create a communication gateway between the Bluetooth/WiFi mesh users and the goTenna mesh users.

**Comment by NF8C** [comment]
Firechat might be alright.  I'm not a huge fan based on my personal testing.  The trick would be to allow users to turn that feature on and off.  It can be a battery hog.  I love the idea of a custom protocol but that's a lot of work.  What would be incredible is if Gottena would extend their protocol to include Bluetooth comms between phones.

**Comment by TechExaminer** [comment]
I agree, Firechat definitely needs major improvements. I also agree the ability to turn the feature on and off would be critically important. It would be awesome if goTenna implemented these features directly into there preexisting app but I’m not sure that will ever happen. If we want that to happen I think a lot more people would need to start emailing them requesting these features as well as leaving reviews about it in the App Store.

**Comment by dsuresh-ap** [comment]
This sounds like a cool idea. Unfortunately we will not be adding integrations into our SDK. It would be great if someone could use our SDK and integrate 3rd parties such as Firechat and Bridgefy!

**Comment by Dorson** [comment]
My guess for how it could be done in steps and practically.

This needs to be **in a library**, where functions are well defined and documented.

Like Node.js but for the decentralized networks.

Individual coders can dedicate to create functions with their defined parameter variables.

It's hard to code everything from scratch for one app alone.

The most hard work is to define the style of the library. Where the most basic functions are defined.

Like the:
+ encode_data:pacakge ( data, parameters, if any )
+ decode_data_pacakge ( data, parameters, if any )
+ send_data ( data, parameters, if any )
+ seek_peers ( parameters, if any )
+ update_peers ( update list of peers , parameters, if any )
+ privacy_jump ( call to seek new peers, if conditions are met )

etc....

Node.js could be an inspiration for how to try to create such a library.

**Maybe the TOR, the Onion, IPFS and Torrent apps for the browser did do most of the work for that already.**

Lots of things to copy from previous efforts. 
Maybe a few fun puzzles to solve for the people who like network and security puzzles.
